src/Sarif.Viewer.VisualStudio/ResultTextMarker.cs

```diff
@@ -61,15 +89,37 @@ public ResultTextMarker(int runIndex, Region region, string fullFilePath)
         /// <returns>Returns true if a VS editor was opened.</returns>
         public bool TryNavigateTo(bool usePreviewPane)
         {
-#pragma warning disable VSTHRD010 // This is an analyzer false positive. 
-            return this.TryNavigateTo(usePreviewPane, retryNaviation: true);
-#pragma warning restore VSTHRD010
-        }
-
-        private bool TryNavigateTo(bool usePreviewPane, bool retryNaviation)
-        {
             ThreadHelper.ThrowIfNotOnUIThread();
 
+            // If the tag doesn't have a persistent span, or it isn't open
+            // then this indicates that we need to attempt to open the document.
+            if (this.tag?.DocumentPersistentSpan.Span == null ||
+                !this.tag.DocumentPersistentSpan.IsDocumentOpen ||
+                this.tag.DocumentPersistentSpan.Span.TextBuffer == null)
```

Comment:
Add a helper since this check is performed (although inverted) below so the check is consistent